systemtests/src/test/java/io/enmasse/systemtest/bases/IoTTestBase.java

```diff
@@ -135,4 +146,29 @@ protected void createIoTProject(IoTProject project) throws Exception {
         TimeMeasuringSystem.stopOperation(operationID);
     }
 
+    protected void waitForFirstSuccessOnTelemetry(HttpAdapterClient adapterClient) throws Exception {
+        waitForFirstSuccess(adapterClient, MessageType.TELEMETRY);
+    }
+
+    protected void waitForFirstSuccess(HttpAdapterClient adapterClient, MessageType type) throws Exception {
+        JsonObject json = new JsonObject(Map.of("a", "b"));
+        TestUtils.waitUntilCondition("First successful "+type.name().toLowerCase()+" message", () -> {
+            try {
+                if(type == MessageType.EVENT) {
+                    var response = adapterClient.sendEvent(json, any());
```

Comment:
Can we add some logging here to see better what's going on when messages are not being processed properly?